sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -155,11 +158,13 @@ export class EducationProgramOfferingController extends BaseController {
       offeringId,
     );
     if (!offering) {
-      throw new UnprocessableEntityException(
+      throw new NotFoundException(
         "Not able to find a Education Program Offering associated with the current Education Program, Location and offering.",
       );
     }
-    return transformToProgramOfferingDto(offering);
+    const hasExistingApplication =
```

Comment:
Is there anything preventing these methods to be executed in parallel?

Reply:
No. I can change to parallel execution.